xarray/core/utils.py

```diff
@@ -549,6 +549,11 @@ def __eq__(self, other) -> bool:
     def __hash__(self) -> int:
         return hash((ReprObject, self._value))
```

Comment:
For the sake of coherency:
```suggestion
        return hash((type(self), self._value))
```